trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1101,6 +1105,83 @@ private Response getTree(UUID expUUID, String outputId, QueryParameters queryPar
         }
     }
 
+    /**
+     * Query the provider for a report. Based on the given output, this
+     * endpoint will return a specific report, which can be an image, a
+     * textual data, a HTML, etc.
+     *
+     * @param expUUID
+     *            desired experiment UUID
+     * @param outputId
+     *            Output ID for the data provider to query
+     * @return {@link Response} with the corresponding report data
+     */
+    @GET
+    @Path("/report/{outputId}")
+    @Tag(name = RPT)
+    @Produces({ MediaType.APPLICATION_OCTET_STREAM, MediaType.APPLICATION_JSON })
```

Comment:
Do we need MediaType.APPLICATION_JSON? In what cases will it return a JSON reply?